File: src/sysinfo/cpu/cpu.cpp
#include "./cpu.h"
#include <fstream>
#include <regex>

std::string cpu() {
  std::ifstream cpuinfo("/proc/cpuinfo");
  std::string line;
  std::regex model_name_regex(R"(model name\s*:\s*(.*))");
  std::smatch match;

  while (std::getline(cpuinfo, line)) {
    if (std::regex_search(line, match, model_name_regex)) {
      return match[1];
    }
  }

  return " ";
}

File: src/sysinfo/cpu.rs
use std::fs::File;
use std::io::Read;

pub fn get_cpu() -> String {
    let mut file = File::open("/proc/cpuinfo").unwrap();
    let mut temp = String::new();
    file.read_to_string(&mut temp).unwrap();

    for line in temp.lines() {
        if let Some(cpu) = line.strip_prefix("model name") {
            let cpu = cpu.replace(": ", "");
            return cpu.trim().to_string();
        }
    }
    "Cpu not found".to_string()
}
